cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go

```diff
@@ -56,58 +54,66 @@ func NewControlPlanePreparePhase() workflow.Phase {
 	}
 }
 
-func getControlPlanePreparePhaseFlags() []string {
-	return []string{
+func getControlPlanePreparePhaseFlags(name string) []string {
+	flags := []string{
 		options.APIServerAdvertiseAddress,
 		options.APIServerBindPort,
 		options.CfgPath,
 		options.ControlPlane,
 		options.NodeName,
-		options.FileDiscovery,
-		options.TokenDiscovery,
-		options.TokenDiscoveryCAHash,
-		options.TokenDiscoverySkipCAHash,
-		options.TLSBootstrapToken,
-		options.TokenStr,
 	}
+	if name != "manifests" {
```

Comment:
do we need these flags for the certs sub phase?

Reply:
yes, because certs should be executed before kubeconfig (and so before admin.conf exists)